### DISCUSSION TITLE
custom domain-specific rule languages

### DISCUSSION THREAD

**Discussion by linkjaya** [evrete/evrete#37] [Q&A]
Is there an example of custom domain-specific rule languages in evrete? I am looking for way to load external knowledge file in if else format.
Example 
if color = red then flower = [ rose, lily]
If color = yellow then flower = [ tulip, orchid ]

**Comment by andbi** [comment]
The very nature of domain-specific languages is that they need to be translated into the rule engine's format, and for a rule engine there's no way of foreseeing of making provisions for all the possible language formats. 

In your case, you need a parser that would read your source line by line as described here: https://github.com/evrete/evrete/discussions/36#discussioncomment-4788161
